src/yupSchema.js

```diff
@@ -74,9 +78,65 @@ function getRequiredErrorMessage(inputType, { inlineError, configError }) {
 
 const getJsonTypeInArray = (jsonType) =>
   Array.isArray(jsonType)
-    ? jsonType.find((val) => val !== 'null') // eg ["string", "null"] // optional fields - get the lead type.
+    ? jsonType.find((val) => val !== 'null') // eg ["string", "null"] // optional fields - get the head type.
     : jsonType; // eg "string"
 
+const getOptionsAllowed = (field) => {
+  const onlyValues = field.options?.map((option) => option.value);
+  const optionsBroken = [
+    ...onlyValues,
+    '', // [1]
+  ];
+
+  if (field.required) {
+    return [
+      ...optionsBroken,
+      null, // [2]
+    ];
+  }
+
+  const isOptionalWithNull =
+    Array.isArray(field.jsonType) &&
+    // @TODO should also check the "oneOf" directly looking for "null"
+    // option but we don't have direct access at this point.
+    // Otherwise the JSON Schema validator will fail as explained in PR#18
+    field.jsonType.includes('null');
+  if (isOptionalWithNull) {
+    return [...optionsBroken, null];
+  }
+
+  return optionsBroken;
```

Comment:
based on my confusion on the internal pipelines failures. What are the instances where a null is not required?

Reply:
fwiw i did test this by just always including `null` and `''` in the `optionsBroken` and it works .... but am I fully convinced its the way? no

Reply:
> What are the instances where a null is not required?

Do you mean "allowed" instead of "required" right?

- In the ideal world: null is never allowed unless the field explicitly has an option with `const: null`.
- Nowadays: It must be always allowed, in the 3 scenarios:
  - required fields - allowed so the `.required()` works correctly saying "Required field" instead of "Option null is not valid".
  - optional fields (conventional)  - allowed so that we don't break existing integrations that send null when the field is empty.
  - optional fields (null) -  allowed (correct) because it has an option with `const: null`.

Reply:
ok cool, so in that case .. for this one:

> optional fields (conventional) - allowed so that we don't break existing integrations that send null when the field is empty.

we either want to always include it or have a way to check when its conditionally required and don't include when it doesn't match that condition?

Reply:
Looking at our internal APIs, revealed a demo that is failing: [nested conditional fields](https://json-schema-form.vercel.app/?path=/story/combinations--nested-conditional-fields) - [local dev link](http://localhost:6007/?path=/story/combinations--nested-conditional-fields).

P.S. This is not a "nested conditional field". it's a "field with nested conditionals" - the field is not nested, it's normal. The conditional is the one that it's nesting. English is hard :sweat_smile: 

<img width="1376" alt="image" src="https://github.com/remoteoss/json-schema-form/assets/14869087/7d30ada1-8839-4d24-94ae-67eba437a628">

This error happens because the JSON Schema is wrong — `state` is an optional field. At Remote, as we treat empty fields as `null`, the field is missing an option with `const: null`. 

The same bugs happens in the other demos (and probably in our internal JSON Schemas too). If this json schema was a real one, our Remote API would fail with the typical "expected a string, got null" error.

For this reason, you are right, we'll need to keep allowing `null` always for now. And later on, create a migration-strategy:
- 1. in JSF log a warning when we spot these nulls as exceptions, as they should be an error instead.
- 2. in JSF add a feature-flag to allow devs to toggle it and start using the new version without these exceptions.
- 3. Use it ourselves to spot our incorrect json schemasand fix our own UI Form integration.